pimd/pim_upstream.c

```diff
@@ -2001,6 +2001,9 @@ static bool pim_upstream_kat_start_ok(struct pim_upstream *up)
 		return false;
 
 	pim_ifp = ifp->info;
+	if (!pim_ifp)
```

Comment:
This is the wrong fix.  We should stop the KAT on a no ip pim

Reply:
This problem stems from the fact that we do a pim_upstream_del and probably have a wrong ref count somewhere and have left timers running.  I would suspect that we need to go through and stop these timers on instance delete instead of relying on upstream_del ( or fix the coutner issue )

Reply:
Yes, i think so too. Anyway this fix is necessary for that `go through`.

Reply:
>  I would suspect that we need to go through and stop these timers on instance delete instead of relying on upstream_del ( or fix the coutner issue )
After double-check, all are right including counter which is used for `go through`, all are kept even 'no ip pim'.
So it is a simple issue of NULL pointer, and this fix is right and enough.